File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>tmap;
        for(int i=0;i<nums.size();i++){
            int rem=target-nums[i];
            if(tmap.find(rem)!=tmap.end()){
                return {tmap[rem],i};
            }
            tmap.insert({nums[i],i});
        }
        return {};
    }
};

File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        mp ={}
        for i, v in enumerate(nums):
            need = target - v
            if need in mp:
                return [mp[need],i]
            mp[v] = i
        return []

        